src/hooks/useCds/translate.js

```diff
@@ -249,11 +321,52 @@ export function translateResponse(patientData, stridesData) {
   console.log("translate completed.")
 }
 
-function patientDataToHash(patientData) {
-  if (patientData == null || patientData.length === 0) {
-    return {}
+export function translateToggleChange(patientData, toggleStatus) {
+  if (patientData == null || patientData.length == 0) {
+    return;
+  }
+
+  const patient = patientData.find(pd => pd.resourceType == 'Patient');
+  handleToggles(patient, patientData, toggleStatus.isImmunosuppressed, immunosuppressedObservation);
+  handleToggles(patient, patientData, toggleStatus.isPregnant, pregnantObservation);
+  handleToggles(patient, patientData, toggleStatus.isPregnantConcerned, pregnantConcernedObservation);
+  handleToggles(patient, patientData, toggleStatus.isSymptomatic, symptomaticObservation);
+}
+
+function handleToggles(patient, patientData, isChecked, obs) {
+  const newObsId = `new-observation-for-${obs.label.toLowerCase()}`;
+  if (isChecked) {
+    const found = patientData.find(pd => pd.id === newObsId);
+    if (!found) {
+      const newObs = {
```

Comment:
I think it would be best if the Observations have a `status`. This is a required element, and our CDS does filtering based upon the status of Observations.

Reply:
This is still outstanding.

Reply:
See[ comment above](https://github.com/ccsm-cds-tools/ccsm-cds-dashboard/pull/32#discussion_r1506283721). Pregnancy toggle switch will not work without `status` field.

Reply:
Assigned status=final